### PR TITLE
feat: Adding a how-to for deploying UPF in DPDK mode

### PR DESCRIPTION
**Pull request by gatici**
# Description

The implementation of the DPDK support in the UPF has been introduced in https://github.com/canonical/sdcore-upf-operator/pull/57. 
However, documentation of this feature was missing.
Hence, this PR adds a how-to about using UPF in DPDK mode.

# Checklist:

- [x] My code follows the [style guidelines](/CONTRIBUTING.md) of this project
- [x] I have performed a self-review of my own code
- [x] I have made corresponding changes to the documentation
- [ ] I have added tests that validate the behaviour of the software
- [ ] I validated that new and existing unit tests pass locally with my changes
- [ ] Any dependent changes have been merged and published in downstream modules
- [ ] I have bumped the version of the library

### CODE REVIEW

**Review by gruyaume — COMMENTED** [review]
Overall this looks good, I just have a note related to installing Terraform. I haven't tested it though and I assume you have.

File: docs/how-to/deploy_sdcore_user_plane_in_dpdk_mode.md [hunk 1]
```diff
@@ -0,0 +1,156 @@
...
+  name: sriovdp-config
+  namespace: kube-system
+data:
+  config.json: |
+    {
+      "resourceList": [
+        {
+          "resourceName": "intel_sriov_vfio_access",
+          "selectors": {
+            "pciAddresses": ["0000:00:06.0"]
```
Comment (markbeierl) [inline docs/how-to/deploy_sdcore_user_plane_in_dpdk_mode.md:68]: There are different types of selectors. Should we be describing them?

File: docs/how-to/deploy_sdcore_user_plane_in_dpdk_mode.md [hunk 2]
```diff
@@ -0,0 +1,156 @@
...
+Create a Juju model named `user-plane`:
+
+```shell
+juju add-model user-plane user-plane-cloud
+```
+
+Install Terraform.
+
+```shell
+sudo snap install terraform --classic
```
[the remark this answers is not in the document]
Comment (gatici) [reply docs/how-to/deploy_sdcore_user_plane_in_dpdk_mode.md:107]: Yes, we need to add it.

[the remark this answers is not in the document]
Comment (gatici) [reply docs/how-to/deploy_sdcore_user_plane_in_dpdk_mode.md]: Done

**Comment by gatici** [comment]
> Overall this looks good, I just have a note related to installing Terraform. I haven't tested it though and I assume you have.

Terraform is added to requirements. This is tested through the [tutorial] (https://github.com/canonical/charmed-aether-sd-core/pull/12).

**Review by gruyaume — APPROVED** [review]

**Pull request merged**